core/schema/container.go

```diff
@@ -463,6 +463,13 @@ func (s *containerSchema) Install() {
 				),
 			),
 
+		dagql.NodeFunc("withSymlink", DagOpContainerWrapper(s.srv, s.withSymlink)).
+			Doc(`Return a snapshot with a symlink`).
+			Args(
+				dagql.Arg("target").Doc(`Location of the file or directory to link to (e.g., "/existing/file").`),
+				dagql.Arg("linkName").Doc(`Location where the symbolic link will be created (e.g., "/new-file-link").`),
+			),
```

Comment:
Probably makes sense to also take optional `expand`, `permissions` and `owner` args, similar to `withNewFile`: https://github.com/dagger/dagger/blob/54a7d54ad79bb38bb0111086510bb4844d9a0b86/core/schema/container.go#L381-L388

Reply:
Adding `expand` makes sense; however permissions seems like a no go, from `man chmod`:

```
       chmod never changes the permissions of symbolic links; the chmod system
       call cannot change their permissions.  This is not a problem since  the
       permissions  of  symbolic links are never used.  However, for each sym‐
       bolic link listed on the command line, chmod changes the permissions of
       the pointed-to file.
```

However, apparently BSD does support it:

```
       -h      If the file is a	symbolic link, change the mode of the link it-
	       self rather than	the file that the link points to.
```




`chown` on the other hand does support ownership via the `--no-dereference` flag, so there's probably an OS call we can make to support ownership.